cpp/src/parquet/arrow/generate_fuzz_corpus.cc

```diff
@@ -0,0 +1,118 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// A command line executable that generates a bunch of valid Parquet files
+// containing example record batches.  Those are used as fuzzing seeds
+// to make fuzzing more efficient.
+
+#include <cstdlib>
+#include <iostream>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "arrow/array.h"
+#include "arrow/io/file.h"
+#include "arrow/record_batch.h"
+#include "arrow/result.h"
+#include "arrow/table.h"
+#include "arrow/testing/random.h"
+#include "arrow/util/io_util.h"
+#include "parquet/arrow/writer.h"
+
+namespace arrow {
+
+using ::arrow::internal::CreateDir;
+using ::arrow::internal::PlatformFilename;
+
+static constexpr int32_t kBatchSize = 1000;
+static constexpr int32_t kChunkSize = kBatchSize * 3 / 8;
+
+Result<std::shared_ptr<RecordBatch>> ExampleBatch1() {
+  random::RandomArrayGenerator gen(42);
+  std::shared_ptr<Array> a, b, c, d;
+  a = gen.Int16(kBatchSize, -10000, 10000, /*null_probability=*/0.2);
+  b = gen.Float64(kBatchSize, -1e10, 1e10, /*null_probability=*/0.0);
+  c = gen.String(kBatchSize, 0, 30, /*null_probability=*/0.2);
+  {
+    auto values = gen.Int64(kBatchSize * 10, -10000, 10000, /*null_probability=*/0.2);
+    auto offsets = gen.Offsets(kBatchSize + 1, 0, static_cast<int32_t>(values->length()));
+    RETURN_NOT_OK(ListArray::FromArrays(*offsets, *values, default_memory_pool(), &d));
+  }
```

Comment:
Add an integer column with a constant value to trigger RLE.

Reply:
Is that automatic too?

Reply:
I think so, I haven't checked how to trigger it. A repeated constant should, hopefully!